src/wesnoth.cpp

```diff
@@ -779,6 +780,14 @@ static int do_gameloop(const std::vector<std::string>& args)
 
 	game_config_manager config_manager(cmdline_opts);
 
+	if(game_config::check_migration) {
+		game_config::check_migration = false;
+		gui2::dialogs::migrate_version_selection::execute();
+		gui2::show_message("",
```

Comment:
Probably better to put this in the prompt itself, especially since there's a case where the dialog may not show at all.